core/src/main/java/org/apache/iceberg/hadoop/HadoopMetricsContext.java

```diff
@@ -111,6 +113,56 @@ public void increment(Integer amount) {
     };
   }
 
+  /**
+   * The Hadoop implementation delegates to the FileSystem.Statistics implementation and therefore
+   * does not require support for operations like unit() as the counter values are not directly
+   * consumed.
+   *
+   * @param name name of the metric
+   * @param unit ignored
+   * @return counter
+   */
+  @Override
+  public org.apache.iceberg.metrics.Counter counter(String name, Unit unit) {
```

Comment:
We need to deprecate the other counter methods in this class, right?

Reply:
Oh, I think that they already are deprecated because this comes from `MetricsContext`.

Reply:
yep, sorry forgot to reply here but they are already deprecated as you've already noticed :)